### DISCUSSION TITLE
WhichKey conflicting keymaps for mini.comment

### DISCUSSION THREAD

**Discussion by JacquesCherbuin** [LazyVim/LazyVim#1097] [Q&A] [answered]
In the mini.comment plugin there are 3 mappings:

    -- Toggle comment (like `gcip` - comment inner paragraph) for both
    -- Normal and Visual modes
    comment = 'gc',

    -- Toggle comment on current line
    comment_line = 'gcc',

    -- Define 'comment' textobject (like `dgc` - delete whole comment block)
    textobject = 'gc',

'gc' conflicts with 'gcc' and :checkhealth and shows this warning:

![which-key-conflict](https://github.com/LazyVim/LazyVim/assets/63825183/6f893dc3-c8ac-4a22-ba72-6e31ee917f1f)

When I tried to override the mappings in key lua/plugins/ , :checkhealth still shows the conflict.

I want to map comment_line to 'gl'.

Anyone know how to fix this conflict?

**Comment by jyuan0** [comment, marked as answer]
For me, `:checkhealth` doesn't show a 'gc' conflict if `comment_line` is changed to `gl`:
```
  {
    "echasnovski/mini.comment",
    opts = {
      mappings = {
        comment = "gc",
        comment_line = "gl",
        textobject = "gc",
      },
    },
  },
```

**Reply by JacquesCherbuin** [reply]
Oh I didn't realise that it can be changed in opts, I was trying to change in keys. Now it works. Thank you

**Reply by dpetka2001** [reply]
In keys you usually change keys that are not mapped internally by a plugin. If a plugin comes with internal mappings then it's often in the `opts` section under a `mappings` or something else that the plugin has configured in its default config. Reading the docs for the default config of the plugin in question helps you understand how nested you should go in `opts` to change the configuration.

**Reply by JacquesCherbuin** [reply]
Ok that makes sense. Thanks

**Reply by ssh352** [reply]
still shows conflict with this config.

```
WhichKey: checking conflicting keymaps
- WARNING conflicting keymap exists for mode **"n"**, lhs: **"gc"**
- rhs: `
```

**Comment by pkazmier** [comment]
Out of curiosity, should an out-of-the-box installation of LazyVim show any conflicting keymap warnings? I just noticed this same warning on a new install. Should we post a bug report?

**Reply by folke** [reply]
It's a warning, don't worry about it
